features/staging_commands.feature

```diff
@@ -542,3 +542,35 @@ Feature: Enable command behaviour when attached to an UA staging subscription
            | release | fips-apt-source                                                        |
            | xenial  | https://esm.staging.ubuntu.com/fips-updates/ubuntu xenial-updates/main |
            | bionic  | https://esm.staging.ubuntu.com/fips-updates/ubuntu bionic-updates/main |
+
+
+    @series.lts
+    @uses.config.machine_type.lxd.vm
+    Scenario Outline: Commands when cloud init didn't run properly
+        Given a `<release>` machine with ubuntu-advantage-tools installed
+        When I delete the file `/run/cloud-init/instance-data.json`
+        And I attach `contract_token_staging` with sudo
+        Then stdout matches regexp:
+        """
+        Warning: cloud-init is present but did not run. Assuming not on cloud.
```

Comment:
The message here could be wrong right ? The `instance-data.json` file could not be generated because cloud-init failed on some of the earlier stages of the machine.

However, this is a corner case, so don't know if we should take it into consideration. What do you think @mitechie ?

Reply:
Could the message be something like `did not run properly`?

Reply:
It could, but then we would take away the case that cloud-init was disabled by the user

Reply:
If it was disabled by the user, then it did not run - makes sense imho

Reply:
In my understanding we are talking about either:

* `cloud-init is present but did not run.`: This will cover the case that cloud-init is disabled, but fail to cover when it actually failed to run.
* `cloud-init did not run properly`: Which will cover the failed scenario but not the disabled one.

But I can be missing something about the message change you are proposing

Reply:
Side note: the actual case that brought this to our attention recently was while building an image: cloud-init has not run _yet_.

How about `cloud-init is present but has not detected the cloud`. I think that is still accurate for all the scenarios where this could occur